app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt

```diff
@@ -29,125 +32,191 @@ object EditPatientEffectHandler {
+    val patientWithoutAgeOrDateOfBirth = patient.copy(
+        fullName = ongoingEditPatientEntry.name,
+        gender = ongoingEditPatientEntry.gender,
+        dateOfBirth = null,
+        age = null
+    )
+
     return when (ongoingEditPatientEntry.ageOrDateOfBirth) {
-      is OngoingEditPatientEntry.EitherAgeOrDateOfBirth.EntryWithAge -> {
-        patient.copy(
-            fullName = ongoingEditPatientEntry.name,
-            gender = ongoingEditPatientEntry.gender,
-            dateOfBirth = null,
-            age = coerceAgeFrom(patient.age, ongoingEditPatientEntry.ageOrDateOfBirth.age, utcClock)
-        )
+      is EntryWithAge -> {
+        val age = coerceAgeFrom(patient.age, ongoingEditPatientEntry.ageOrDateOfBirth.age, utcClock)
+        patientWithoutAgeOrDateOfBirth.copy(age = age)
       }
-      is OngoingEditPatientEntry.EitherAgeOrDateOfBirth.EntryWithDateOfBirth -> {
-        patient.copy(
-            fullName = ongoingEditPatientEntry.name,
-            gender = ongoingEditPatientEntry.gender,
-            age = null,
-            dateOfBirth = LocalDate.parse(ongoingEditPatientEntry.ageOrDateOfBirth.dateOfBirth, dateOfBirthFormatter)
-        )
+
+      is EntryWithDateOfBirth -> {
+        val dateOfBirth = LocalDate.parse(ongoingEditPatientEntry.ageOrDateOfBirth.dateOfBirth, dateOfBirthFormatter)
+        patientWithoutAgeOrDateOfBirth.copy(dateOfBirth = dateOfBirth)
       }
     }
   }
 
-  private fun coerceAgeFrom(
-      alreadySavedAge: Age?,
-      enteredAge: String,
-      utcClock: UtcClock
-  ): Age {
+  private fun coerceAgeFrom(alreadySavedAge: Age?, enteredAge: String, utcClock: UtcClock): Age {
     val enteredAgeValue = enteredAge.toInt()
-
     return when {
-      // When prefilling the details, the age text changed event will get triggered, which will
-      // trigger an update of the age and the age updated at timestamp which will change the age
-      // calculations again. This handles that case.
       alreadySavedAge != null && alreadySavedAge.value == enteredAgeValue -> alreadySavedAge
       else -> Age(enteredAgeValue, Instant.now(utcClock))
     }
   }
+
+  private fun updateAddress(
+      patientAddress: PatientAddress,
+      ongoingEditPatientEntry: OngoingEditPatientEntry
+  ): PatientAddress {
+    return patientAddress.copy(
+        colonyOrVillage = ongoingEditPatientEntry.colonyOrVillage,
+        district = ongoingEditPatientEntry.district,
+        state = ongoingEditPatientEntry.state
+    )
+  }
+
+  private fun savePhoneNumber(
+      savePatientEffects: Observable<SavePatientEffect>,
+      patientRepository: PatientRepository
+  ): Observable<EditPatientEvent> {
+    val effectsWithPhoneNumber = savePatientEffects
+        .map { (entry, patient, _, phoneNumber) -> Triple(patient.uuid, phoneNumber, entry.phoneNumber) }
+        .filter { (_, existingPhoneNumber, enteredPhoneNumber) ->
+          existingPhoneNumber != null || enteredPhoneNumber.isNotBlank()
```

Comment:
Can we extract this conditional check `existingPhoneNumber != null || enteredPhoneNumber.isNotBlank()` into a separate function and give it a meaningful name?